models/Element/AbstractElement.php

```diff
@@ -137,6 +137,40 @@ public function setPath($path)
      */
     protected ?int $userModification = null;
 
+    /**
+     * @internal
+     *
+     * @var int|null
+     */
+    protected ?int $parentId = null;
+
+
+    /**
+     * @return int|null
+     */
+    public function getParentId()
+    {
+        return $this->parentId;
+    }
+
+    /**
+     * @param int $parentId
+     * @return $this
+     */
+    public function setParentId($parentId)
+    {
+        $parentId = (int) $parentId;
+        $type = Service::getElementType($this);
+
+        if ($parentId != $this->parentId && $type === 'object') {
```

Comment:
please avoid that kind of ifs in the parent classes ... if there is specific functionality of a subclass, then it should go into the subclass...

Reply:
Fixed by overriding the `setParentId` method in `AbstractObject` class